.circleci/config.yml

```diff
@@ -169,7 +172,6 @@ workflows:
           requires:
             - build
           filters:
-            branches:
-              only: master
             tags:
-              only: /^v?[0-9]+(\.[0-9]+)*$/
+              # match tags such as "v1.1.1" or "anything-stage"
+              only: /^v?[0-9]+(\.[0-9]+)*$|(?:-stage)$/
```

Comment:
Based on the circle docs I'm not sure this will work. See https://circleci.com/docs/2.0/workflows/#using-regular-expressions-to-filter-tags-and-branches.

If the tag was `foo-stage` this regex only matches the suffix. Since their setup needs an exact match, you'll need to extend the regex to match anything ahead of the stage suffix. See the example given in their docs:

> For example, only: /^config-test/ only matches the config-test tag. To match all tags starting with config-test, use only: /^config-test.*/ instead.

I think to fix this we could also keep the numeric prefix so the regex could be:

`^v?[0-9]+(\.[0-9]+)*(?:-stage)?$`

This should then completely match all of the following:

- v1.2
- 1.2
- v1.2-stage
- 1.2-stage

Reply:
Hmm, this is strange, because it matched this branch, i.e. `tag-pushing-stage` fine, and ran the deployment step to push to -stage, as can be seen https://circleci.com/gh/mozilla/extension-workshop/537?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

I guess we can do what you proposed to be sure.

Reply:
That seems wrong doesn't it? I wouldn't expect a deploy to happen for a PR branch? Is that running because there's now no branch filter?

Reply:
Heh, I guess this branch name inadvertently showed that it didn't take an exact match for the tag filter to work. However, I think you're right that branches should not trigger deployment no matter what their names are.

I've updated the PR with comments. Please review again.